### ISSUE TITLE
GCM Push to iOS and Android in same request (not really an issue, but need some info)

### ISSUE THREAD

**Issue by mvestuto** [phonegap/phonegap-plugin-push#1303]
Noob, here - sorry if this is in the wrong spot. Not sure where I should have posted this question.

Before I get too far, from what I gather - this is not really an issue and things are working as expected but I would like some education on GCM, this plugin and the differences in how android reacts with "data", and ios requires "notification" payloads. I ultimately want to send bulk notifications to all registered devices in one large request. I have android and iOS versions of my app, and for efficiency sake would like to send to both platforms at the same time.

 In experimentation, it looks like I need to send one request to android devices and a separate request to ios devices due to the differences in payloads and androids reaction to a "mixed" payload.

I'm using phonegap build and GCM to send notifications. I can succesfully send and receive android notifications in foreground/background using "data" in the payload. I can succesfully send and receive ios notifications in the foreground/background using notification in the payload.

Here's my android payload

"data": {
			"title": "Summer Babies Notification",
			"body": "Geoffrey just posted in chat",
			"channelId": "7",
			"content-available": 1,
			"sound":"default"
		}

Here's my iOS payload
"notification": {
			"title": "Summer Babies Notification",
			"body": "Geoffrey just posted in chat",
			"channelId": "7",
			"content-available": 1,
			"content_available": "true",
			"sound": "default"
		}

So it seems (from experimentation and documentation) iOS devices won't display notifications when receiving the data payload. If I send both "data" and "notification" listed below the .on('notification', function(data) event is fired in iOS but not on android in background notifications. It's not until I remove "notification" from the request that android behaves as expected when tapping on a background notification.

"registration_ids":["xxxx-android","xxxx-iOS"],
		"data": {
			"title": "Summer Babies Notification",
			"body": "Geoffrey just posted in chat",
			"channelId": "7",
			"content-available": 1,
			"sound":"default"
		},
		"notification": {
			"title": "Summer Babies Notification",
			"body": "Geoffrey just posted in chat",
			"channelId": "7",
			"content-available": 1,
			"content_available": "true",
			"sound": "default"
		}

What happens when I tap on the background notification in the shade, is the app opening another instance of itself, or "some sort" of similar behavior. When I'm inspecting the device in gap debug - another entry for the app appears in the list. Not sure of the technical terminology for that one.

In short, is it possible to push one GCM notification to both platforms at the same time - and have the .on('notification.. event fire reliably. Do people do this sort of thing, or should I just use separate calls to GCM for each platform?

**Comment by macdonst** [comment]
@mvestuto no, it is not currently possible to push on GCM notification and have it display properly on both Android and iOS. As you've noticed, Android has to have everything in `data`. If the payload is in `notification` then the OS takes over and the app never gets notified of a new notification. On iOS in order to use GCM everything has to be sent via the `notification` section.
